wsd/FileServer.cpp

```diff
@@ -1116,14 +1114,30 @@ void FileServerRequestHandler::preprocessFile(const HTTPRequest& request,
                 << "frame-ancestors " << frameAncestors;
         std::string escapedFrameAncestors;
         Poco::URI::encode(frameAncestors, "'", escapedFrameAncestors);
-        Poco::replaceInPlace(preprocess, std::string("%FRAME_ANCESTORS%"), escapedFrameAncestors);
+        replaceVar["FRAME_ANCESTORS"] = escapedFrameAncestors;
     }
     else
     {
         LOG_TRC("Denied all frame ancestors");
         cspOss << imgSrc << "; ";
     }
 
+    std::string newpreprocess;
+    newpreprocess.reserve(preprocess.length() + 50);
+    std::stringstream checkVar(preprocess);
+    std::string medium;
+
+     while(getline(checkVar, medium, '$'))
```

Comment:
Isn't the variable marker `%` ? Perhaps a comment to explain what this loop is doing is in order.